tests/zfs-tests/tests/functional/cli_root/zfs_mount/zfs_mount_006_pos.ksh

```diff
@@ -94,17 +96,31 @@ log_must $ZFS create $TESTPOOL/$TESTFS1
 unmounted $TESTPOOL/$TESTFS1 || \
 	log_must force_unmount $TESTPOOL/$TESTFS1
 
-while [[ -n $mtpt ]] ; do
-	(( depth == MAXDEPTH )) && \
+while (( depth >= 0 )) ; do
+	if (( depth == MAXDEPTH )) ; then
 		log_note "Verify that '$ZFS $mountcmd <filesystem>' " \
 		"which mountpoint be the identical of an existing one " \
-		"will fail with return code 1."
-
-	log_must $ZFS set mountpoint=$mtpt $TESTPOOL/$TESTFS1
-	log_mustnot $ZFS $mountcmd $TESTPOOL/$TESTFS1
-
-	unmounted $TESTPOOL/$TESTFS1 || \
-		log_fail "Filesystem $TESTPOOL/$TESTFS1 is mounted."
+		"will fail with return code 1, but it will succeed on linux."
+		if is_linux; then
+			log_must $ZFS set mountpoint=$mtpt $TESTPOOL/$TESTFS1
+			log_must $ZFS $mountcmd $TESTPOOL/$TESTFS1
+			
+			mounted $TESTPOOL/$TESTFS1 || \
+				log_fail "Filesystem $TESTPOOL/$TESTFS1 is unmounted."
+		else
+			log_must $ZFS set mountpoint=$mtpt $TESTPOOL/$TESTFS1
+			log_mustnot $ZFS $mountcmd $TESTPOOL/$TESTFS1
+			
+			unmounted $TESTPOOL/$TESTFS1 || \
+				log_fail "Filesystem $TESTPOOL/$TESTFS1 is mounted."
+		fi
+	else
+		$ZFS set mountpoint=$mtpt $TESTPOOL/$TESTFS1
```

Comment:
@behlendorf @BearBabyLiu 
Thanks for your review. All input is welcomed here. I need to update my master for  testing. I will continue  to focus on this script